### PR TITLE
Aqua.jl

### PR DESCRIPTION
**Pull request by kunzaatko**
Minor tweak to Aqua.jl configuration in `runtests.jl` and add badge to README

- edit(Agua): Update issue link and mark as broken
- docs(README): Add `Agua.jl` badge

### CODE REVIEW

**Comment by jishnub** [comment]
Thanks, I didn't know about the `broken` flag

**Comment by kunzaatko** [comment]
I limited the compat of `Aqua` as you suggested to `v0.7`. That was a good catch. I also switched the plugin to the new format of `Project.toml` without the extra target. It is more organized that way I think and does not make any practical difference because the compat for julia is not pre`v1.2` where the change occured.

**Comment by jishnub** [comment]
Let's not change the format. Although this should work fine in principle, there are occasional subtle conflicts between the main `Manifest.toml` and the `test/Manifest.toml` files that are hard to debug, especially when developing the package. The older format doesn't have these issues.

**Comment by jishnub** [comment]
It seems, using the `broken` flag prints out the entire list of ambiguities when running the tests. I'm not sure if this is desired.

**Comment by kunzaatko** [comment]
I reverted it to the previous format.
I guess the idea behind the `broken` flag showing the output is, that it is broken only temporarily for a short time, and it works as a reminder... The decision is up to you though.

**Comment by kunzaatko** [comment]
Is there something that I should know about the new format? Is there a reason why I shouldn't be using it in my packages if they have compat `julia=^1.2`?

**Comment by jishnub** [comment]
I liked the idea of the `broken` flag, but perhaps `Aqua` should summarise the broken tests instead of printing them all. This is ultimately an upstream decision, though

**Comment by jishnub** [comment]
> Is there something that I should know about the new format? Is there a reason why I shouldn't be using it in my packages if they have compat julia=^1.2?

I can't find the link immediately, but there used to be a warning that stated that the interplay between the `Manifest.toml` and `test/Manifest.toml` is still not fully worked out (and IIRC there was a `Pkg` PR about the `test` environment inheriting the main one instead of re-resolving). In the past, I've occasionally faced issues when the main and test environments load different versions of the same package, and I've found that using [`TestEnv.jl`](https://github.com/JuliaTesting/TestEnv.jl) makes the process of instantiating the test environment much smoother.

File: Project.toml [hunk 1]
```diff
@@ -30,4 +30,12 @@ Statistics = "10745b16-79ce-11e8-11f9-7d13ad32a3b2"
 Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
 
 [targets]
-test = ["Aqua", "Test", "Base64", "ReverseDiff", "SparseArrays", "StaticArrays", "Statistics"]
```
Comment (jishnub) [inline Project.toml:33]: Looks like `Aqua` expects this to be on a single line

Comment (kunzaatko) [reply Project.toml]: Now it should be fine

**Comment by kunzaatko** [comment]
> In the past, I've occasionally faced issues when the main and test environments load different versions of the same package

There is a test for differences between the test environment and the main environment versions. So testing with `Aqua.jl` should eliminate this.

**Comment by jishnub** [comment]
The Aqua test checks that the same dependencies are listed in the `[targets]` section and the `test/Project.toml`. I was referring to the case where these two are identical, but the difference is in the m
. Anyway

**Comment by jishnub** [comment]
For future reference, I was referring to this issue: https://github.com/JuliaLang/Pkg.jl/issues/1585, but perhaps this is fixed now

**Pull request merged**